src/trainer.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * This software is licensed under the terms of the MIT License.
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2011-2019, Lukas Weber <laochailan@web.de>.
+ * Copyright (c) 2012-2019, Andrei Alexeyev <akari@taisei-project.org>.
+ */
+
+#include "taisei.h"
+
+#include "trainer.h"
+
+#include "global.h"
+#include "stage.h"
+
+
+void trainer_init(Trainer *tnr) {
+	memset(tnr, 0, sizeof(Trainer));
+	// store two sets of values
+    // one is "per stage"
+    // the other is for the entire game session
+    tnr->trainer_lives_stage = 0;
+    tnr->trainer_bombs_stage = 0;
+    tnr->trainer_hits_stage = 0;
+
+    tnr->trainer_lives_total = 0;
+    tnr->trainer_bombs_total = 0;
+    tnr->trainer_hits_total = 0;
+}
+
+
+bool trainer_enabled(void) {
+    if (config_get_int(CONFIG_TRAINER_MODE) && config_get_int(CONFIG_TRAINER_STATS)) {
```

Comment:
I'm a bit confused about the `CONFIG_TRAINER_STATS` setting. What is it for exactly?

Reply:
It determines whether or not the stats are shown in real-time on the HUD, or kept strictly to the Stage Clear screen - I'm sure some people might find them distracting. However, its usage here wasn't quite correct, but I've fixed it.